lib/controllers/widget/detailed_device.dart

```diff
@@ -0,0 +1,237 @@
+// limitations under the License.
+
+import 'package:chrome_management_app/controllers/devices.dart';
+import 'package:chrome_management_app/controllers/widget/custom_fields_card.dart';
+import 'package:chrome_management_app/controllers/widget/hardware_os_card.dart';
+import 'package:chrome_management_app/models/fade_on_scroll_util.dart';
+import 'package:chrome_management_app/models/error_handler.dart';
+import 'package:chrome_management_app/objects/basic_device.dart';
+import 'package:chrome_management_app/objects/detailed_device.dart';
+import 'package:chrome_management_app/views/login.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:http/http.dart' as http;
+import 'package:intl/intl.dart';
+
+/// Stateful Widget to show the summary of a detailed device.
+class DetailedDeviceSummary extends StatefulWidget {
+  /// Authorization token to get information from the Directory API.
+  final String _authToken;
+
+  /// Http Client
+  final http.Client _client;
+
+  final BasicDevice _device;
+
+  /// Constructor of Widget
+  DetailedDeviceSummary(this._authToken, this._client, this._device, {Key key})
+      : super(key: key);
+  @override
+
+  /// Create initial state of the Widget.
+  _DetailedDeviceSummaryState createState() =>
+      _DetailedDeviceSummaryState(_authToken, _client, _device);
+}
+
+/// DeviceList Widget states
+class _DetailedDeviceSummaryState extends State<DetailedDeviceSummary> {
+  /// Authorization token to get information from the Directory API.
+  String _authToken;
+
+  /// Http Client
+  http.Client _client;
+
+  /// Verify the device is loading.
+  bool _loading = false;
+
+  /// Stores any error/exception code.
+  ErrorHandler _warning;
+
+  /// Verify if a error/exception happend while loading the list of devices.
+  bool _errorOnLoading = false;
+
+  /// Basic information of the device.
+  BasicDevice _basicDevice;
+
+  /// Full information of the device.
+  DetailedDevice _fullDevice;
+
+  /// Scroll controller to manage animations.
+  ScrollController _scrollController = ScrollController();
+  _DetailedDeviceSummaryState(this._authToken, this._client, this._basicDevice);
+
+  /// Initial state of the Widget.
+  @override
+  void initState() {
+    _scrollController = ScrollController();
+    super.initState();
+    _getDetailedDevice();
+  }
+
+  /// Get initial page of the list of devices
+  _getDetailedDevice() async {
+    setState(() {
+      _loading = true;
+    });
+    Devices.getDetailedDevice(_client, _authToken, _basicDevice.deviceId)
+        .then((value) {
+      setState(() {
+        _fullDevice = value;
+        _loading = false;
+      });
+    }).catchError((e) {
+      setState(() {
+        _loading = false;
+        _errorOnLoading = true;
+        _warning = e;
+      });
+    });
+  }
+
+  /// Pops an alert to user and sends the app to the login view.
+  _alertAndLogIn(ErrorHandler error) {
```

Comment:
All of these alert functions shouldn't be here, they should be in a centric point and shared between all code. Maybe you can put them in the ErrorHandler

Reply:
I already move them to Error Handler class, thanks for the advice.